File: my-code/119/main.go
package main

import "fmt"

// @Time     : 2021/08/07 13:03
// @Author   : Ranshi
// @File     : main.go

func getRow(rowIndex int) []int {
	row := make([]int, rowIndex+1)
	row[0] = 1
	for i := 1; i <= rowIndex; i++ {
		for j := i; j > 0; j-- {
			row[j] += row[j-1]
		}
	}
	return row
}

func main() {
	fmt.Println(getRow(3))
}

File: my-code/119/main.py
# -*- coding:utf-8 -*-
# @Time     : 2021/05/28 18: 00
# @Author   : Ranshi
# @File     : 119. 杨辉三角 II.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        res = [1, 1]
        if rowIndex == 0:
            return [1]
        for i in range(2, rowIndex + 1):
            res = (
                [1]
                + [res[j] + res[j + 1] for j in range(0, len(res) - 1)]
                + [1]
            )
        return res


if __name__ == "__main__":
    s = Solution()
    print(s.getRow(4))
